web/hooks/use-saved-contract-metrics.ts

```diff
@@ -1,33 +1,28 @@
 import { ContractMetric } from 'common/contract-metric'
 import { Contract } from 'common/contract'
 import { usePersistentState, storageStore } from './use-persistent-state'
-import { useUser, useUserContractMetrics } from './use-user'
-import { useEffectCheckEquality } from './use-effect-check-equality'
+import { useUser } from './use-user'
 import { safeLocalStorage } from 'web/lib/util/local'
+import { getUserContractMetrics } from 'common/supabase/contract-metrics'
+import { db } from 'web/lib/supabase/db'
+import { useEffect } from 'react'
 
 export const useSavedContractMetrics = (contract: Contract) => {
   const user = useUser()
-  const contractMetrics = contract
-    ? useUserContractMetrics(user?.id, contract.id)
-    : null
+
   const [savedMetrics, setSavedMetrics] = usePersistentState<
     ContractMetric | undefined
   >(undefined, {
     key: `contract-metrics-${contract.id}`,
     store: storageStore(safeLocalStorage),
   })
 
-  const metrics =
-    contractMetrics || savedMetrics
-      ? ({
-          ...savedMetrics,
-          ...contractMetrics,
-        } as ContractMetric)
-      : undefined
-
-  useEffectCheckEquality(() => {
-    if (metrics) setSavedMetrics(metrics)
-  }, [metrics, setSavedMetrics])
+  useEffect(() => {
+    getUserContractMetrics(user?.id ?? '_', contract.id, db).then((metrics) => {
```

Comment:
I think this is not live updating? Don't we need that for the case of making a bet from the market page and seeing your position update in the BetSummary?

Reply:
We might be able to pass the current user's position down from the ContractPage if that has loaded all users' positions. And maybe other parts of the app don't need this to live update...

Reply:
Well, on the contract page, this is only used for the leaderboard, which only shows if it's resolved, meaning it doesn't have to live update and one fetch should suffice